### PR TITLE
Sortingview curation widget

### PR DESCRIPTION
**Pull request by alejoe91**
- Implement sortingview curation widget (example [curation link](https://figurl.org/f?v=gs://figurl/spikesortingview-10&d=sha1://1ba03f81e62ec7cb2e3e46898830f92cdf5e026f&label=SpikeInterface%20-%20Sorting%20Summary)):
```
sw.plot_sorting_summary(we, curation=True, backend="sortingview")
```

- The curation view allows users to create a link associated to a curation. The `spikeinterface.curation.apply_sortingview_curation(sorting, uri)` function enables one to curate an existing SI object with the manual curation from the link (using the `CurationSorting` class) (example [curated link](https://figurl.org/f?v=gs://figurl/spikesortingview-10&d=sha1://1ba03f81e62ec7cb2e3e46898830f92cdf5e026f&label=SpikeInterface%20-%20Sorting%20Summary&s={%22sortingCuration%22:%22jot://cENcosqWTArO%22}})):
```
from spikeinterface.curation import apply_sortingview_curation
sorting_view_uri = "jot://deeVxsBTDxad"
sorting_curated = apply_sortingview_curation(sorting, sorting_view_uri)
```

- Added a small function to `delete_property()`

### CODE REVIEW

File: spikeinterface/curation/sortingview_curation.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    except ImportError:
+        raise ImportError(
+            "To apply a SortingView manual curation, you need to have sortingview installed: "
+            ">>> pip install sortingview"
+        )
+    curation_sorting = CurationSorting(sorting, make_graph=False, properties_policy="keep")
+
+    # get sorting view curation
+    try:
+        sorting_curation = kcl.load_json(uri=uri)
```
Comment (samuelgarcia) [inline spikeinterface/curation/sortingview_curation.py:48]: This variavle name is very very ambiguous `curation_sorting` is already used for `CurationSorting`
Here it is a simple dict.
I would call it `curation_dict`

File: spikeinterface/curation/sortingview_curation.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    except:
+        raise Exception(f"Could not retrieve curation from SortingView uri: {uri}")
+
+    # first, merge groups
+    if "mergeGroups" in sorting_curation:
+        merge_groups = sorting_curation["mergeGroups"]
+        for mg in merge_groups:
+            if verbose:
+                print(f"Merging {mg}")
+            curation_sorting.merge(mg, new_unit_id="-".join(mg))
```
Comment (samuelgarcia) [inline spikeinterface/curation/sortingview_curation.py:58]: Here you force the new ID to be a string but what happen if the old unit_ids where just in ?

File: spikeinterface/curation/sortingview_curation.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    # gather and apply properties
+    labels_dict = sorting_curation["labelsByUnit"]
+    properties = {}
+    for _, labels in labels_dict.items():
+        for label in labels:
+            if label not in properties:
+                properties[label] = np.zeros(len(curation_sorting.current_sorting.unit_ids), dtype=bool)
```
Comment (samuelgarcia) [inline spikeinterface/curation/sortingview_curation.py:66]: Why doing this. The results from sorting view do have differents property per units ?
The dict is not the same acrros units ?
You should add a line to explain

Comment (alejoe91) [reply spikeinterface/curation/sortingview_curation.py]: added comment

File: spikeinterface/curation/sortingview_curation.py (hunk 2 above)
Comment (alejoe91) [reply spikeinterface/curation/sortingview_curation.py]: good point! chekcing unit dtype now and only doing it for int

File: spikeinterface/curation/sortingview_curation.py (hunk 1 above)
Comment (alejoe91) [reply spikeinterface/curation/sortingview_curation.py]: renamed `sortingview_curation_dict`

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**